pygdf/categorical.py

```diff
@@ -47,11 +46,14 @@ def __init__(self, dtype, codes_dtype, categories, ordered):
         self._codes_impl = numerical.NumericalSeriesImpl(codes_dtype)
 
     def __eq__(self, other):
-        if isinstance(other, CategoricalSeriesImpl):
-            return all([self.dtype == other.dtype,
-                        tuple(self._categories) == tuple(other._categories),
-                        self._ordered == other._ordered,
-                        self._codes_impl == other._codes_impl])
+        try:
+            return (isinstance(other, CategoricalSeriesImpl) and
+                    self.dtype == other.dtype and
+                    tuple(self._categories) == tuple(other._categories) and
+                    self._ordered == other._ordered and
+                    self._codes_impl == other._codes_impl)
+        except Exception:
+            return False
```

Comment:
Because you can't compare pandas custom dtypes and numpy dtypes. We hide the exception so that you can compare categorical and numeric implementations. Although in this case we can skip the try block because we check if other is `CategoricalSeriesImpl`. The try/except for `NumericalSeriesImpl` is still needed though.